pkg/aws/aws_client/role.go

```diff
@@ -89,3 +96,202 @@ func (client *AWSClient) ListRoles() ([]types.Role, error) {
 	out, err := client.IamClient.ListRoles(context.TODO(), input)
 	return out.Roles, err
 }
+
+func (client *AWSClient) PolicyAttachedToRole(roleName string, policyArn string) (bool, error) {
+	policies, err := client.ListRoleAttachedPolicies(roleName)
+	if err != nil {
+		return false, err
+	}
+	for _, policy := range policies {
+		if *policy.PolicyArn == policyArn {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func (client *AWSClient) ListRoleAttachedPolicies(roleName string) ([]types.AttachedPolicy, error) {
```

Comment:
NP : reword func name to match client call, or be more specific to call out we are listing attached policies by role name
```suggestion
func (client *AWSClient) ListAttachedRolePolicies(roleName string) ([]types.AttachedPolicy, error) {
```